### ISSUE TITLE
validator registrations with lighthouse

### ISSUE THREAD

**Issue by xenowits** [ObolNetwork/charon#2106]
# 🎯 Problem to be solved

We want to support [validator registrations](https://ethereum.github.io/beacon-APIs/#/Validator/registerValidator) with lighthouse Validator Client.

Ciaran already worked on such a [PR](https://github.com/sigp/lighthouse/pull/3445) 7 months back. But since then, we have formalized the solution with this [design doc](https://docs.google.com/document/d/1J9Kv2LNZtJk_4rG88UGs5ngtL7d14suSsnlgZ5KPcKA/edit#heading=h.mssb0822zapb). We have also added an internal proposal for a new [endpoint](https://github.com/ObolNetwork/beacon-APIs/pull/4) `/eth/v1/validator/proposer_config` to standardize how different VCs fetch override values for `pubkey` & `timestamp`.

So, in a nutshell, we want lighthouse VC to be capable of fetching override values from charon DVT and use those values to perform validator registrations in a DV setting.

# 🛠️ Proposed solution

* Fix the outdated PR with a new PR branching off latest unstable branch of `sigp:lighthouse`. Use the original PR as a reference.
* Add support for the new endpoint `/eth/v1/validator/proposer_config`  to lighthouse VC

# 🧪 Tests
- [ ] Tested by new automated unit/integration/smoke tests
- [ ] Manually tested on core team/canary/test clusters
- [ ] Manually tested on local compose simnet

**Comment by xenowits** [comment]
See PR on lighthouse - https://github.com/sigp/lighthouse/pull/4306

**Comment by boulder225** [comment]
@boulder225 to add comments

**Comment by KaloyanTanev** [comment]
Not relevant anymore
